ui/packages/markdown/src/Markdown.svelte

```diff
@@ -1,14 +1,28 @@
 <script lang="ts">
 	import { createEventDispatcher } from "svelte";
 	import "./typography.css";
-
 	export let elem_id: string = "";
 	export let visible: boolean = true;
 	export let value: string;
 	export let min_height = false;
+	import { afterUpdate } from "svelte";
 
 	const dispatch = createEventDispatcher<{ change: undefined }>();
 
+	let target: HTMLElement;
+	let katex: Module | undefined;
+	afterUpdate(async () => {
+		let latex_elements = target.querySelectorAll(".math");
+		if (latex_elements.length === 0) {
+			return;
+		}
+
+		if (katex === undefined) {
+			katex = await import("./katex");
+		}
+		katex.render(latex_elements);
+	});
```

Comment:
This only needs to happen when the `value` changes rather than when _any_ props change.

Pull the body of this into a function and use a reactive statement:

```ts
aysnc function handle_katex() {
  let latex_elements = target.querySelectorAll(".math");
  if (latex_elements.length === 0) {
    return;
  }

  if (katex === undefined) {
    katex = await import("./katex");
  }
  katex.render(latex_elements);
}
```

And:

```ts
$: value && handle_katex();
```

I'd also test this with some more complex blocks. You may need to `await tick()` at the very beginning, to ensure changes have been flushed to the DOM.

Reply:
now rendering is server-side, but good to know!